programs/ziti-edge-tunnel/instance.c

```diff
@@ -401,6 +401,34 @@ tunnel_identity_array get_tunnel_identities() {
     return tnl_id_arr;
 }
 
+tunnel_identity_array get_tunnel_identities_for_metrics() {
+
+    tunnel_identity_array arr = get_tunnel_identities();
+    tunnel_identity_array tnl_id_arr = NULL;
```

Comment:
`get_tunnel_identities` never returned NULL. Will callers of this function be prepared for NULL?

Reply:
yes, the caller is checking for NULL. I added a free(arr) at end of this function now, because get_tunnel_identities() create a new arr array